iFuture/src/containers/FeedPage/index.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux'
+import { push } from 'connected-react-router'
+import { routes } from '../../containers/Router'
+
+import * as S from './styles'
+import { TextField, InputAdornment } from '@material-ui/core'
+
+import Appbar from '../../components/Appbar';
+import CardRestaurant from '../../components/CardRestaurant';
+import Bottombar from '../../components/Bottombar';
+import { getRestaurants, getActiveOrder } from '../../actions/food';
+import OrdersInProgress from '../../components/OrdersInProgress';
+
+function FeedPage(props) {
+
+  const { goToSearch, restaurants, getRestaurants, order, getActiveOrder } = props
+
+  useEffect(() => {
+    getRestaurants()
+    getActiveOrder()
+  }, [getRestaurants, getActiveOrder])
+
+  const [categoryClicked, setCategoryClicked] = useState(false)
+  const [categorySelected, setCategorySelected] = useState('')
+
+  const selectCategory = (category) => {
+    setCategoryClicked(true)
+    setCategorySelected(category)
+    if (category === categorySelected) {
+      setCategoryClicked(!categoryClicked)
+      setCategorySelected('')
+    }
```

Comment:
Para evitar "piscadas" onde valores mudam, eu englobaria essa lógica inicial dentro do else. Do jeito que o código está quando `category === categorySelected` for verdadeiro, ele primeiro vai setar (`setCategoryClicked`) para `true` e em seguida `!categoryClicked`, sugestão:

```js
    if (category === categorySelected) {
      setCategoryClicked(!categoryClicked)
      setCategorySelected('')
    } else {
       setCategoryClicked(true)
       setCategorySelected(category)
    }
```